### ISSUE TITLE
GradientExplainer only works on copies of models?

### ISSUE THREAD

**Issue by delacylab** [shap/shap#2929]
I am trying to use shap with an LSTM and data with 4 time periods so my datasets typically look like (num_rows,4,num_features). Because DeepExplainer is not compatible with Tensorflow 2.x, I have been using GradientExplainer but ran into this bizarre problem where calling the Explainer with my model gives me a warning but calling it on a copy of the model works properly.
_clf = Sequential()
clf.add(Bidirectional(LSTM(layer_size, activation='tanh', return_sequences=True))) 
clf.add(Bidirectional(LSTM(layer_size, activation='tanh')))
clf.add(Dense(2, activation = 'softmax'))
clf.compile(loss="categorical_crossentropy", optimizer = tfa.optimizers.AdamW(weight_decay=.01), metrics=['accuracy'])
y_cat = tf.keras.utils.to_categorical(y, num_classes=2) 
early_stopping_monitor = EarlyStopping(min_delta = 0.01, monitor='loss',patience=10)
clf.fit(X,y_cat, epochs=200, callbacks=[early_stopping_monitor])
e = shap.GradientExplainer(clf, X)_

If I then later call:
_shap_values, shap_indexes = e.shap_values(rep)_ 
I get this warning:
WARNING:tensorflow:Layers in a Sequential model should only have a single input tensor, but we receive a <class 'list'> input: [<tf.Tensor: shape=(138, 4, 187)...

I do not get the warning if I change the final line in the first code segment to:
_e = shap.GradientExplainer(copy.deepcopy(clf), X)_
Has anyone else run into this? Is shap thread-safe or is there something else going on? This is the only way I have got GradientExplainer to work without warnings that worry me,

**Comment by CloseChoice** [comment]
Thanks for reporting the issue. Could you please provide a minimal reproducible example for this? So, specifying how to get `X` and `y` in your code or using a dummy dataset?
Otherwise it is hard for us to look deeper into this.
